app/src/main/java/com/janewaitara/businesscardapp/MainActivity.kt

```diff
@@ -1,12 +1,253 @@
+            BusinessCard(
+                namesArray[2],
+                addressesArray[2],
+                contactsArray[2],
+                emailsArray[2],
+                portfoliosArray[2],
+                companiesArray[2],
+                imageList[2]
+            ),
+            BusinessCard(
+                namesArray[3],
+                addressesArray[3],
+                contactsArray[3],
+                emailsArray[3],
+                portfoliosArray[3],
+                companiesArray[3],
+                imageList[3]
+            ),
+            BusinessCard(
+                namesArray[4],
+                addressesArray[4],
+                contactsArray[4],
+                emailsArray[4],
+                portfoliosArray[4],
+                companiesArray[4],
+                imageList[3]
+            ),
+            BusinessCard(
+                namesArray[5],
+                addressesArray[5],
+                contactsArray[5],
+                emailsArray[5],
+                portfoliosArray[5],
+                companiesArray[5],
+                imageList[0]
+            )
+        )
+
+
+        change_bs_card_button.setOnClickListener { view ->
+            getRandomBusinessCard()
+            displayNewCard()
+        }
+
+        if (savedInstanceState != null){
+            currentName = savedInstanceState.getString(MEMBER_NAME_KEY)!!
+            currentAddress = savedInstanceState.getString(ADDRESS_KEY)!!
+            currentCompany = savedInstanceState.getString(COMPANY_KEY)!!
+            currentContact = savedInstanceState.getString(CONTACT_KEY)!!
+            currentEmail = savedInstanceState.getString(EMAIL_KEY)!!
+            currentPortfolio = savedInstanceState.getString(PORTFOLIO_KEY)!!
+            currentImage = savedInstanceState.getInt(IMAGE_KEY)
+
+            Log.d("Image", currentImage.toString())
+
+        } else{
+
+            getRandomBusinessCard()
+        }
+
+        displayNewCard()
+
+    }
+    /**
+     * Saving the Business Card properties to maintain the same card when the orientation of a screen changes
+     * */
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+
+        outState.putString(MEMBER_NAME_KEY,currentName)
+        outState.putString(ADDRESS_KEY,currentAddress)
+        outState.putString(COMPANY_KEY,currentCompany)
+        outState.putString(CONTACT_KEY,currentContact)
+        outState.putString(EMAIL_KEY,currentEmail)
+        outState.putString(PORTFOLIO_KEY,currentPortfolio)
+        outState.putInt(IMAGE_KEY,currentImage)
+
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        super.onCreateOptionsMenu(menu)
+        menuInflater.inflate(R.menu.menu,menu)
+        return true
     }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        super.onOptionsItemSelected(item)
+        when (item.itemId){
+            R.id.share_menu -> showShareOptions()
```

Comment:
Excellent job on the share dialog and displaying the build version with the menu.